docs/extensions.adoc

```diff
@@ -706,7 +705,9 @@ methods in the interface, so that only the needed ones need to be overridden.
   like attaching interceptors to various interception points as described in the chapter <<Interceptors>>.
 
 `stop()`::
-  This is called once at the very end of the Spock execution.
+  This is called at least once at the very end of the Spock execution.
+
+NOTE: Build tools might trigger repeated executions, so those methods might be called more often than you might expect.
```

Comment:
https://github.com/spockframework/spock/issues/1240 -> https://issues.apache.org/jira/browse/SUREFIRE-1864

And we currently have the jvm shutdown hook for `stop`.

Reply:
Ah, I remember that issue.
But the questions basically remain.
Is this called on different instances or the same instance?
It is the same for `start`, isn't it? Because you only change the text for `stop`.
If it is on the same instance can we maybe somehow proxy it to only call the actual method only once?
No matter which way it is, we need to document how it is exactly, for example is it "start start start stop stop stop", or is it "start stop start stop start stop"? And is it on separate instances or on different ones? Otherwise it will be pretty hard to correctly implement those methods, a mere "they might be called more often than you might expect" is way too vague to be any helpful and only increases confusion.
It should maybe not be part of this PR but more in the scope of #1240 though.

Reply:
I've removed the contentious note, IIRC JUnit Platform implement a workaround that reduces Surefire's bad behavior.

Reply:
Sorry to stay stubborn, but now it is still saying for `start` "once" and for `stop` "at least once".
Shouldn't it at least be "at least once" for both until it is maybe documented more extensively? :-)

Reply:
Last open point here :-)

Reply:
If we are talking about a single instance, then no `start` will only be called once for the instance.